src/Fixer/AttributeNotation/GeneralAttributeRemoveFixer.php

```diff
@@ -0,0 +1,210 @@
+use PhpCsFixer\Tokenizer\Analyzer\NamespaceUsesAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Raffaele Carelle <raffaele.carelle@gmail.com>
+ *
+ * @implements ConfigurableFixerInterface<_AutogeneratedInputConfiguration, _AutogeneratedComputedConfiguration>
+ *
+ * @phpstan-import-type _AttributeItems from AttributeAnalysis
+ * @phpstan-import-type _AttributeItem from AttributeAnalysis
+ *
+ * @phpstan-type _AutogeneratedInputConfiguration array{
+ *  attributes?: list<string>
+ * }
+ * @phpstan-type _AutogeneratedComputedConfiguration array{
+ *  attributes: list<string>
+ * }
+ */
+final class GeneralAttributeRemoveFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /** @use ConfigurableFixerTrait<_AutogeneratedInputConfiguration, _AutogeneratedComputedConfiguration> */
+    use ConfigurableFixerTrait;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Configured attributes should be omitted (FQNs strings).',
+            [
+                new CodeSample(
+                    '<?php
+#[\A\B\Foo]
+function foo() {}
+',
+                    ['attributes' => ['\A\B\Foo']]
+                ),
+                new CodeSample(
+                    '<?php
+use A\B\Bar as BarAlias;
+
+#[\A\B\Foo]
+#[BarAlias]
+function foo() {}
+',
+                    ['attributes' => ['\A\B\Foo', 'A\B\Bar']]
+                ),
+            ]
+        );
+    }
+
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return \defined('T_ATTRIBUTE') && $tokens->isTokenKindFound(T_ATTRIBUTE);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        if (0 === \count($this->configuration['attributes'])) {
+            return;
+        }
+
+        $index = 0;
+
+        while (null !== $index = $tokens->getNextTokenOfKind($index, [[T_ATTRIBUTE]])) {
+            $attributeAnalysis = AttributeAnalyzer::collectOne($tokens, $index);
+
+            if (1 === \count($attributeAnalysis->getAttributes())) {
+                $elements = [
+                    [
+                        'name' => $attributeAnalysis->getAttributes()[0]['name'],
+                        'start' => $attributeAnalysis->getStartIndex(),
+                        'end' => $attributeAnalysis->getEndIndex(),
+                    ],
+                ];
+            } else {
+                $elements = array_map(static fn (array $attribute): array => [
+                    'name' => $attribute['name'],
+                    'start' => $attribute['start'],
+                    'end' => $attribute['end'],
+                ], $attributeAnalysis->getAttributes());
+            }
+
+            $endIndex = $attributeAnalysis->getEndIndex();
+
+            $removedCount = 0;
+            foreach ($elements as $element) {
```

Comment:
Remapping to `$elements` is superfluous, it does not bring any value. `AttributeAnalysis::getAttributes()` returns array of attributes' representation with `start`, `end` and `name`, so the remapping basically proxies all the data 1:1. All tests pass when I remove this and iterate over analysis' results:

```suggestion
            $endIndex = $attributeAnalysis->getEndIndex();

            $removedCount = 0;
            foreach ($attributeAnalysis->getAttributes() as $element) {
```

Reply:
Good. Fixed.